tests/unit_tests/job_queue/test_lsf_driver.py

```diff
@@ -75,75 +121,216 @@ def mock_bjobs(tmp_path):
+    # print("JOBID USER STAT QUEUE FROM_HOST EXEC_HOST JOB_NAME SUBMIT_TIME")
+    Path(dir / "mocked_result").write_text(
+        f"{job_id}\tpytest\t{status}\ttest_queue\thost1\thost2\ttest_job\t{str(datetime.now())}"
     )
 
-    os.chmod(script_path, 0o755)
 
+async def test_submit_failure_script_exit(mock_bsub, caplog, tmpdir, monkeypatch):
+    monkeypatch.chdir(tmpdir)
+    lsf_driver = LSFDriver(None)
+    lsf_driver.set_option("BSUB_CMD", tmpdir / "mock_bsub")
+    lsf_driver._max_attempt = 3
+    lsf_driver._SLEEP_PERIOD = 0.2
+    with patch(
+        "ert.job_queue.realization_state.QueueableRealization"
+    ) as mock_realization:
+        mock_realization.realization.run_arg.iens.return_value = 0
+        mock_realization.realization.job_script = "exit.sh"
+        mock_realization.realization.run_arg.runpath = "/usr/random/ert_path"
+        with pytest.raises(
+            RuntimeError, match="Maximum number of submit errors exceeded"
+        ):
+            await lsf_driver.submit(mock_realization)
+
+    job_paths = Path("job_paths").read_text(encoding="utf-8").strip().split("\n")
+
+    # should try command 3 times before exiting
+    assert len(job_paths) == 3
+
+    output = caplog.text
+    assert re.search("Tried submitting job 3 times, but it still failed", output)
+    assert len(re.findall("returned non-zero exitcode: 1", output)) == 3
+
+
+async def test_submit_success(mock_bsub, caplog, tmpdir, monkeypatch):
+    caplog.set_level(logging.DEBUG)
+    monkeypatch.chdir(tmpdir)
+    lsf_driver = LSFDriver(None)
+    lsf_driver.set_option("BSUB_CMD", tmpdir / "mock_bsub")
+    lsf_driver._max_attempt = 3
+
+    with patch(
+        "ert.job_queue.realization_state.QueueableRealization"
+    ) as mock_realization:
+        mock_realization.realization.run_arg.iens.return_value = 0
+        mock_realization.realization.job_script = "valid_script.sh"
+        mock_realization.realization.run_arg.runpath = "/usr/random/ert_path"
+        await lsf_driver.submit(mock_realization)
+
+    job_paths = Path("job_paths").read_text(encoding="utf-8").strip().split("\n")
+    assert len(job_paths) == 1
+
+    output = caplog.text
+    print(f"{output=}")
+    assert re.search("Submitted job.*and got LSF JOBID", output)
+    assert re.search("submitted to default queue", output)
+
+
+async def test_poll_statuses_while_already_polling(
+    mock_bjobs, caplog, tmpdir, monkeypatch
+):
+    monkeypatch.chdir(tmpdir)
+
+    lsf_driver = LSFDriver(None)
+    lsf_driver._currently_polling = True
+    lsf_driver._statuses.update({"test_lsf_job_id": "RUNNING"})
+    statuses: dict = await lsf_driver.poll_statuses()
+    assert statuses["test_lsf_job_id"] == "RUNNING"
+    # Should never call bjobs
+    assert not Path("bjobs_logs").exists()
+
+    output = caplog.text
+    assert output == ""
+    assert lsf_driver._currently_polling
+
+
+async def test_poll_statuses_before_submitting_jobs(
```

Comment:
I am not really sure that this is an important feature of the driver, except that it does not crash in the situation.